### DISCUSSION TITLE
Ags widgets on swaylock

### DISCUSSION THREAD

**Discussion by nicefaa6waa** [end-4/dots-hyprland#191] [Q&A] [answered]
Ok alright sorry for wasting your time.Can i replace swaylock with i3lock tho?i mean i know that swaylock is part of swaywm so it won't be possible without breaking lots of stuff but i know people can add eww widgets on i3lock.Here is an example.
https://github.com/dark-dryu/i3lock-widgets-config
I know you don't like eww that much and ags is way better but if it possible i really need to use widgets on lock screen I'd use anything.

**Comment by end-4** [comment, marked as answer]
- it's not about eww or ags; it's about xorg or wayland. this is a major difference.
  - i3lock is only for xorg, while swaylock is a wayland locker. in short, **i3lock does not work in hyprland**
-  eww and ags are the same in this context
- if you want a screen lock that you can spawn widgets  on, use gtklock. it is a layer shell. although it's now broken on hyprland, iirc there's a flag that makes it work in an insecure mode. obviously, i recommend against usin g it this way.

**Reply by nicefaa6waa** [reply]
Thanks a lot.I will try to find another way

**Comment by nicefaa6waa** [comment]
Hm what about an ags widget.Like can i just disable or unassing super + l from swaylock to toggle that ags widget but not like toggle just run once so it won't close if super + l pressed again.Only way to close that ags field is to write the correct password just as cancel button in session widget does

**Reply by end-4** [reply]
You could enter a [submap](https://wiki.hyprland.org/Configuring/Binds/#submaps) to disable all keybinds.
It does sound good, but I'm not sure how well this will work.
